src/cloudant/search.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python
+# Copyright (c) 2016 IBM. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+API module for composing and executing Cloudant Search indexes.
+"""
+from ._common_util import codify
+from cloudant.error import CloudantException
+
+
+class SearchIndex(dict):
+    """
+    Encapsulates a search index as a dictionary based object, exposing the
+    search index function as an attribute and supporting query/data access
+    through the search endpoint.  A Search object is instantiated with a
+    reference to a DesignDocument and is typically used as part of the
+    :class:`~cloudant.design_document.DesignDocument`
+    search index management API.
+    """
```

Comment:
The comment here talks about a `Search` object but the class is `SearchIndex`.  I'm also unclear about the section of docstring that talks about "supporting query/data access through the serch endpoint".  I understand that this was a straight copy of the View class but the View class is callable.  Have you decided that the SearchIndex will be callable as well or is the plan to have a separate class for search.  I know that we spoke a lot about this but I just want to be sure about what your plan is for search index management as it related to search query execution.